src/Microsoft.ApplicationInsights/DataContracts/DependencyTelemetry.cs

```diff
@@ -19,6 +22,7 @@ public sealed class DependencyTelemetry : OperationTelemetry, ITelemetry, ISuppo
 
         internal readonly RemoteDependencyData InternalData;
         private readonly TelemetryContext context;
+        private readonly Lazy<IDictionary<string, object>> operationDetails;
```

Comment:
For compactness (fewer object allocations), I prefer not to use `Lazy<T>`.
Instead, make the field non-readonly `IDictionary<string, object>` and use `System.Threading.LazyInitializer.EnsureInitialized`.

The pattern is this:
```C#
using static System.Threading.LazyInitializer;

private IDictionary<string, object> operationDetails;

public IDictionary<string, object> OperationDetails => EnsureInitialized(ref operationDetails, () => new ConcurrentDictionary<string, object>());
```